src/error/Detail.h

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (C) 1996-2020 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef _SQUID_SRC_ERROR_DETAIL_H
+#define _SQUID_SRC_ERROR_DETAIL_H
+
+#include "base/Here.h"
+#include "base/RefCount.h"
+#include "error/forward.h"
+#include "http/forward.h"
+#include "mem/forward.h"
+#include "sbuf/forward.h"
+
+/// interface for supplying additional information about a transaction failure
+class ErrorDetail: public RefCountable
+{
+public:
+    using Pointer = ErrorDetailPointer;
+
+    virtual ~ErrorDetail() {}
+
+    /// \returns a single "token" summarizing available details
+    /// suitable as an access.log field and similar output processed by programs
+    virtual SBuf brief() const = 0;
+
+    /// \returns all available details; may be customized for the given request
+    /// suitable for error pages and other output meant for human consumption
+    virtual SBuf verbose(const HttpRequestPointer &) const = 0;
+};
+
+/// creates a new NamedErrorDetail object with a unique name
+/// \see NamedErrorDetail::Name for naming restrictions
+ErrorDetail::Pointer MakeNamedErrorDetail(const char *name);
+
+/// system call failure detail based on standard errno(3)/strerror(3) APIs
+class SysErrorDetail: public ErrorDetail {
```

Comment:
Please split all these specialized classes (SysErrorDetail, ExceptionErrorDetail, NamedErrorDetail) into separate files named after the class contained. Since all this code is in a convenience library there is no need to bunch them together like this.

Please also inline the one-liner methods wherever possible.

Reply:
Finally I did not split.
There are source files which need more than one of class definitions this is will result to more files included.
Looks ok  include them from "error/forward.h" so the uses include only this file?
In the other hand these classes are very small, I am not seeing any reason to spread them into separate files.

Reply:
By that logic all Squid code would be in one huge file. Other code combining code is not a good reason for all code to do so. It needs to be considered case-by-case and I believe this is a case where they should be split.

Reply:
commit f62b229a6f

> By that logic all Squid code would be in one huge file

It is not the same.  Splitting STL exceptions will result to about 10 new include files instead of just including "exception". 

In any case, and finally, this change in Squid does not looks bad.